lib/calc_controller.dart

```diff
@@ -3,140 +3,23 @@ import 'package:flutter/material.dart';
 enum Operator { plus, equal }
 
 class CalcState {
-  CalcState(
-      {this.num1 = '0', this.num2 = '', this.result = '0', this.operator});
+  CalcState({this.currentResult = '0'});
 
-  final String num1;
-  final String num2;
-  final String result;
-  final String operator;
+  final String currentResult;
 }
 
 class CalcController {
   final ValueNotifier<CalcState> state = ValueNotifier(CalcState());
 
-  void onNumber(String input) {
-//    printInfo();
-    if (state.value.operator == null) {
-      state.value = CalcState(
-          num1: state.value.num1.replaceAll('0', '') + input,
-          result: state.value.num1.replaceAll('0', '') + input);
-      printInfo();
+  void onNumberPressed(int buttonValue) {
+    final currentResult = state.value.currentResult;
+    if (currentResult == '0') {
```

Comment:
this is not going to work in long term but its fine and satisfies existing tests